src/components/common/block/Block.ts

```diff
@@ -123,18 +126,18 @@ export default class Block<P> {
         } else {
           child.dispatchComponentDidMount();
         }
-      })
-    }
+      });
+    };
     dispatchCDMInner(Object.values(this.children));
   }
 
-  _componentDidUpdate(oldProps: object, newProps: object) {
+  private _componentDidUpdate(oldProps: object, newProps: object) {
```

Comment:
Типы пропсов должны быть типом, который передается в класс
```suggestion
  private _componentDidUpdate(oldProps: P, newProps: P) {
```